### DISCUSSION TITLE
Help with connections issues

### DISCUSSION THREAD

**Discussion by Whalanboy** [nicotine-plus/nicotine-plus#3136] [Q&A]
My ISP uses CG NAT - this means the IP address detected by N+ is not mine and port forwarding doesn't work. I can connect, search and download but nobody can see my shares. Does anybody have any ideas on how I can get around this?

**Comment by slook** [comment]
> nobody can see my shares

That's not exactly true, probably those users who do themselves have an open port can see your shares, because the server makes indirect connections if a direct peer connection fails. Those same users are the only ones who you can get search results and download from.

> how I can get around this?

It is needed to setup a virtual route using some kind of VPN solution. I haven't tried this yet, but I would like to try https://tailscale.com/ to create a distributed proxy relay for this purpose, see links:

https://tailscale.com/use-cases/multi-cloud-networking
https://tailscale.com/blog/how-tailscale-works#nat-traversal
https://tailscale.com/blog/how-nat-traversal-works
